source/_posts/routing-in-react.md

```diff
@@ -0,0 +1,590 @@
+you can turn to that page and start reading. This is similar to how routing works in a React app.
+
+The different stories in the book are like the different pages or components in a React app, and the table of contents
+is like a special tool that helps you move between them. When you click on a link in the table of contents, it takes you
+to the right page you can see the story you want to read.
+
+### React Router
+
+
+React Router enables "client-side routing". This module provides components to handle routing by defining the routes in
+your application and transitioning between them.
+
+Use create react app to quickly create a react app up and running.
+
+```bash
+npx create-react-app my-app
+```
+
+To get started with React Router, you will first need to install the react router package. You can **install React
+Router** from the public npm registry with either npm or yarn in your react application. You can use the following
+command to install react router using npm.
+
+```bash
+npm install react-router-dom
+```
+
+Once the library is installed, you can import the components you want to use in your application. For example, to use
+the **BrowserRouter**, **Route**, **Link,** and other components, you can add the following statement to your code to
+import them from the **React router library**
+
+```bash
+import { BrowserRouter, Route, Link } from 'react-router-dom';
+```
+
+Then, you can use these components to define the routes in your application and specify which components should be
+rendered for each route. Here is a simple example of how this might look:
+
+```javascript
+//import react first
+import React from 'react';
+import {
+    BrowserRouter as Router,
+    Routes,
+    Route,
+    Link
+} from 'react-router-dom';
+
+function App() {
+    return (
+        <Router>
+            <div>
+                <nav>
+                    <ul>
+                        <li>
+                            <Link to="/">Home</Link>
+                        </li>
+                        <li>
+                            <Link to="/about">About</Link>
+                        </li>
+                        <li>
+                            <Link to="/contact">Contact</Link>
+                        </li>
+                    </ul>
+                </nav>
+            </div>
+            <div style={{padding:10}}>
+                <Routes>
+                <Route path={"/"} element={<Home/>}/>
+                <Route path={"/about"} element={<About/>}/>
+                <Route path={"/contact"} element={<Contact/>}/>
+            </Routes>
+
+            </div>
+
+        </Router>
+
+    );
+}
+
+function Home() {
+    return <h2>Home</h2>;
+}
+
+function About() {
+    return <h2>About</h2>;
+}
+
+function Contact() {
+    return <h2>Contact</h2>;
+}
+
+export default App;  
+```
+
+In this example, we are using the **BrowserRouter** component provided by React Router to create a router for our
```

Comment:
- [ ] remove the new line

Reply:
**BrowserRouter** -> `BrowserRouter`
- [ ] use ticks not bolding for syntax highlighting